blockly-core/core/ui/contract_editor/x_button.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+goog.provide('Blockly.XButton');
+
+/**
+ * A DOM-based input which sends events on changes
+ * @param {!Object} options
+ *          onNameChanged (newName: String)
+ *          onEnterPressed
+ * @constructor
+ */
+Blockly.XButton = function (options) {
```

Comment:
`XButton` isn't super clear to me as a name, and the class description above doesn't help much.  Is this a `CloseButton`?  Or something more specific?